### ISSUE TITLE
How can I turn off alarm/shutdown

### ISSUE THREAD

**Issue by redskinhu** [korylprince/hassio-apcupsd#10]
Hello

I have 2 problems.

**1st thing:**
I tried to turn off the alarm/beep of the UPS but it looks unsuccesful.

My Hass.io addon config extra section:
```
  "extra": [
    {
      "key": "BEEPSTATE",
      "val": "N"
    },
    {
      "key": "BATTERYLEVEL",
      "val": "0"
    },
    {
      "key": "MINUTES",
      "val": "0"
    }
  ]
```
I can manage to list the `apcaccess status` command output (Portainer and console):
```
MBATTCHG : 0 Percent
MINTIMEL : 0 Minutes
```

Whitout the "extra" addon config:
```
MBATTCHG : 5 Percent
MINTIMEL : 3 Minutes
```

So two settings are looks ok, I didn't tested yet. But the UPS sill beeps when the power is out.

I saw couple of instruction about the alarm. They said I need to run `apctest` command and disable it. I tried to run it in the console but I got error message because the daemon still running and the `apctest` can't create lock file. I tried to stop the daemon (`/etc/init.d/apcupsd stop` or `service apcupsd` or `systemctl stop apcupsd`) with no luck. (I'm a not a big linux/docker juggler...)

Any idea?

**2nd problem:**

HW/SW config for the clearer view:
I have a Hass.io/APC UPS Daemon on my laptop. The UPS connected to the laptop by USB. The UPS powered the network devices only (router/switch). The laptop have a decent battery so the laptop don't need UPS and don't need to shutdown when the UPS discharged.

How can i turn off the shutdown?

Thanks
T

**Comment by korylprince** [comment]
## Beeping

I'm not 100% sure, but I think `BEEPSTATE` controls the beep in the computer itself. Most computers nowadays don't have built-in beepers, so I don't think this option really does anything.

You are correct that you need to use the `apctest` command to change the UPS firmware settings to disable the UPS beep.

The easiest way to do this is to attach another computer to the UPS temporarily that has acpupsd. If that's not an option, you can "trick" the add-on into doing what you need by following these steps:

1. Change the `type` config option to "net" and restart the add-on
    * This causes the add-on to stay running, but not be connected to the UPS
2. Connect to the container using SSH to the host, Portainer, etc
3. Edit `/etc/apcupsd/apcupsd.conf` and change `UPSTYPE net` to `UPSTYPE usb`
    * We do this so `apctest` can connect correctly to the UPS
    * If you're not comfortable using `vi` to edit the file, you can run `apk add --no-cache nano` to install the more user-friendly `nano` command to edit the file
4. Run `apctest` and make the desired changes
5. Exit the container, change the `type` config option back to "usb", and restart the add-on to clean up

## Shutdown

According to the [docs](http://www.apcupsd.org/manual/manual.html), `BATTERYLEVEL` and `MINUTES` should be set to `-1`, not `0` to disable those timers. In theory, setting those to `-1` should stop your device from shutting down. If that doesn't work, you can override the shutdown script as described in Issue #9. Basically just add a drop-in script for the shutdown command that just does `exit 0`.

**Comment by redskinhu** [comment]
Hello

Thanks for the info. (The shutdown solution under testing.)

I tried to disable the beep to connect the UPS to an another (3) machines with apcupsd. But I had a connection error when I wanted to run `apctest`:

> Checking configuration ...
> sharenet.type = Network & ShareUPS Disabled
> cable.type = USB Cable
> mode.type = USB UPS Driver
> Setting up the port ...
> apctest FATAL ERROR in apctest.c at line 321
> Unable to open UPS device.
>   If apcupsd or apctest is already running,
>   please stop it and run this program again.
> apctest error termination completed


After a hour dig I found why (in the Apcupsd 3.10.6 User's Manual):

> Unlike version 3.8.6, apcupsd version 3.10.x no longer has code to program the EEPROM. Instead we have implemented interactive EEPROM modification code in the apctest program. EEPROM programming must be done with apcupsd stopped so that apctest can access the UPS. In addition, EEPROM programming is currently implemented only for UPSes using the Smart protocol running in serial mode. Perhaps at a later time when the appropriate kernel modifications are standard, we will extend EEPROM programming to USB models.

So it loooks I need to create serial cable for the UPS or I need to be patient until the USB EEPROM programming will be implemented or I need to desolder the beeper from the UPS. The last option looks the easiest/quickest for me. ;-)

Thanks for the help.
T

**Comment by redskinhu** [comment]
I changed the config to this:
```
  "extra": [
    {
      "key": "BATTERYLEVEL",
      "val": "-1"
    },
    {
      "key": "MINUTES",
      "val": "-1"
    }
  ]
```
15% battery level and 30 minutes left when the UPC turned off the laptop (like before)... :-( 

Can I retrieve the log somehow?

Thx
T

**Comment by korylprince** [comment]
What UPS do you have? I have a USB UPS (APC Back-UPS ES 600M1) and I'm able to use `apctest` just fine to disable the alarm. Are you sure you followed the steps correctly?

Also, like I said in my first post, you need to set those values to `-1`, not `0`. If that doesn't work then you need to follow the instructions in the issue I linked to to override the shutdown scripts.

**Comment by redskinhu** [comment]
Sorry my mistake, I copied the old cfg (corrected).
And the values are looks good if I run `apcaccess status`

> MBATTCHG : -1 Percent
> MINTIMEL : -1 Minutes

I have Smart-UPS 750 (SUA750I) USB attached.

I tested with BATTERYLEVEL: 80 and it turns off at 80%. I'll test again with -1 this afternoon. If it isn't working I'll bypass the shutdown script.
Maybe the USB cable??? I don't think because I can get the status.

I installed the apcupsd to the normal shell to an another Linux PC. It connected I get the status but I can't run the `apctest`.

Thanks
T
